crates/cli/commands/src/download.rs

```diff
@@ -0,0 +1,202 @@
+// to avoid overwhelming stdout
+struct DownloadProgress {
+    downloaded: u64,
+    total_size: u64,
+    last_displayed: Instant,
+}
+
+impl DownloadProgress {
+    /// Creates new progress tracker with given total size
+    fn new(total_size: u64) -> Self {
+        Self { downloaded: 0, total_size, last_displayed: Instant::now() }
+    }
+
+    /// Converts bytes to human readable format (B, KB, MB, GB)
+    fn format_size(size: u64) -> String {
+        let mut size = size as f64;
+        let mut unit_index = 0;
+
+        while size >= 1024.0 && unit_index < BYTE_UNITS.len() - 1 {
+            size /= 1024.0;
+            unit_index += 1;
+        }
+
+        format!("{:.2} {}", size, BYTE_UNITS[unit_index])
+    }
+
+    /// Updates progress bar
+    fn update(&mut self, chunk_size: u64) -> Result<()> {
+        self.downloaded += chunk_size;
+
+        // Only update display at most 10 times per second for efficiency
+        if self.last_displayed.elapsed() >= Duration::from_millis(100) {
+            let formatted_downloaded = Self::format_size(self.downloaded);
+            let formatted_total = Self::format_size(self.total_size);
+            let progress = (self.downloaded as f64 / self.total_size as f64) * 100.0;
+
+            print!(
+                "\rDownloading and extracting... {:.2}% ({} / {})",
+                progress, formatted_downloaded, formatted_total
+            );
+            io::stdout().flush()?;
+            self.last_displayed = Instant::now();
+        }
+
+        Ok(())
+    }
+}
+
+/// Adapter to track progress while reading
+struct ProgressReader<R> {
+    reader: R,
+    progress: DownloadProgress,
+}
+
+impl<R: Read> ProgressReader<R> {
+    fn new(reader: R, total_size: u64) -> Self {
+        Self { reader, progress: DownloadProgress::new(total_size) }
+    }
+}
+
+impl<R: Read> Read for ProgressReader<R> {
+    fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
+        let bytes = self.reader.read(buf)?;
+        if bytes > 0 {
+            if let Err(e) = self.progress.update(bytes as u64) {
+                return Err(io::Error::other(e));
+            }
+        }
+        Ok(bytes)
+    }
+}
+
+/// Downloads and extracts a snapshot with blocking approach
+fn blocking_download_and_extract(url: &str, target_dir: &Path) -> Result<()> {
+    let client = reqwest::blocking::Client::builder().build()?;
+    let response = client.get(url).send()?.error_for_status()?;
+
+    let total_size = response.content_length().ok_or_else(|| {
+        eyre::eyre!(
+            "Server did not provide Content-Length header. This is required for snapshot downloads"
+        )
+    })?;
+
+    let progress_reader = ProgressReader::new(response, total_size);
+
+    let decoder = Decoder::new(progress_reader)?;
+    let mut archive = Archive::new(decoder);
+
+    archive.unpack(target_dir)?;
+
+    info!(target: "reth::cli", "Extraction complete.");
+    Ok(())
+}
```

Comment:
I think we're okay with just rolling an async variant of this.

there is 
https://github.com/astral-sh/tokio-tar/
which provides an Archive that can use  AsyncRead as input